catalog/dags/common/requester.py

```diff
@@ -42,12 +43,13 @@ class DelayedRequester:
     delay:   an integer giving the minimum number of seconds to wait
              between consecutive requests via the `get` method.
     headers: a dict that will be passed in all requests, unless overridden
-             by kwargs in specific calls to the get method
+             by kwargs in specific calls to the `get` method
     """
 
-    def __init__(self, delay=0, headers=None):
+    def __init__(self, delay: int = 0, headers: dict | None = None):
+        headers = {} if headers is None else headers
         self._DELAY = delay
-        self.headers = headers or {}
+        self.headers = {"User-Agent": prov.UA_STRING} | headers
```

Comment:
Why do we default here in _addition_ to adding this as a default in the ProviderDataIngester?

Reply:
That was the suggestion of the issue, and I went that route first, but it turns out it wasn't enough to set the default header. I leave it after updating the `ProviderDataIngester` as it will be better if each request from the Calalog goes with this header.

Reply:
😮 That's so strange -- do you mean setting it as the default in `DelayedRequester` did not cover all the existing cases? Are we making requests that don't use the requester somewhere?

If it _can_ be added in only one place, I do agree about adding it in the requester instead of the ProviderDataIngester, since that gets complicated with subclasses that override `__init__` etc.

Reply:
> do you mean setting it as the default in DelayedRequester did not cover all the existing cases? 

Yes, the thing is the `ProviderDataIngester` class passes each time its headers to the `DelayedRequester.get_response_json` method (for the possibility of overriding them) so it wasn't enough to add them to `DelayedRequester` only.

> Are we making requests that don't use the requester somewhere?

It's a possibility, I haven't checked but this isn't the issue here, as I explained above. We should be safe with these changes!